projects/igniteui-angular/src/lib/directives/autocomplete/autocomplete.directive.ts

```diff
@@ -293,31 +293,20 @@ export class IgxAutocompleteDirective extends IgxDropDownItemNavigationDirective
             return;
         }
         this.target.close();
-        this.dropDownOpened$.next();
     }
 
     /**
      * Opens autocomplete drop down
      */
     public open() {
-        if (this.disabled || !this.collapsed) {
+        if (this.disabled || !this.collapsed || this._lastListLength === 0 || this._opening) {
```

Comment:
Yeah, if we're going to to use a flag anyway, this seems like the harder way to do it - since it's harder to manage more flags and it's easier to miss sometimes:
1. Type a repeat letter fast ("dd") that doesn't match on the second char
2. You get the second filter during animation, thus the opened event never fires before the close
3. DD now permanently stuck in closed state 
![image](https://user-images.githubusercontent.com/3198469/91035766-22be0d80-e60f-11ea-980e-1e4fa5adad43.png)


Alternative, that's a bit easier to follow - setup a `shouldBeOpen` flag on `open()` and `close()` and use it during the list change to determine if you need to do something